### ISSUE TITLE
Result parameter type discrepancy in sweep loop

### ISSUE THREAD

**Issue by alnlarsen** [opentap/opentap#1593]
When sweeping parameters, the parameter is added as an additional result to all child steps. 

Sweep loop converts the swept value to a string, even if the value is already IConvertible. The swept value is also added as a result to the sweep loop itself, but here the actual type is used, and not the IConvertible type. This causes issues for some result listeners that do not expect result values to change type.

**Comment by rmadsen-ks** [comment]
@alnlarsen, I think this might be theoretically the correct thing to do, but I think the result listeners should be programmed to handle the situation.

**Comment by alnlarsen** [comment]
@rmadsen-ks I agree that the result listener should handle it correctly, but I also think we should change the behavior in OpenTAP. I think my linked PR has the correct behavior.
